core/embed/lib/error_handling.h

```diff
@@ -20,6 +20,151 @@
 #ifndef LIB_ERROR_HANDLING_H
 #define LIB_ERROR_HANDLING_H
 
+// Suppresses the intellisense error in VSCode
+#ifndef __FILE_NAME__
+#define __FILE_NAME__ __FILE__
+#endif
+
+// Build status code from any 16-bit value.
+// This macro is used for generic status codes.
+#define TS_BUILD(code) ((code) | (~(code) << 16))
```

Comment:
It would be nice to use something that ensures a large minimum Hamming distance. This has 2. However, we don't have to worry about that at this moment.

Reply:
Yes, I agree, I will come up with something more sophisticated.

Reply:
Perhaps we don't need to have a large minimum distance, but only a large distance between `TS_OK` and the error states.

Reply:
see this change cf075b5370487e7a88db7ad3295abd6196c1da4d, now the hamming distance between `TS_OK` and any `TS_ERROR_xxx` is 16. Hamming distance between error codes can be improved as well.